src/base/static/components/translator.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import qs from "qs";
+
+import { LanguageConsumer } from "../context/language";
+
+const Translator = props => {
+  const [translatedMsg, setTranslatedMsg] = useState(props.msg);
+  useEffect(
+    () => {
+      if (
+        props.defaultLanguage.code === props.currentLanguage.code ||
```

Comment:
If the flavor content's default language matches the current language, we can skip translating.

This won't work for user-generated content; but as you mentioned we could provide a button that would render a give piece of user content in the flavor's current language.

Reply:
Though I wonder what the UX of that would be like. Perhaps we could have a button that would say "translate this post and all others to the current language." Then if the user clicks the button all user content by default after that point would be run through Google Translate.

Reply:
I agree we can punt translation of user-generated content for now.

Auto-detection of non-default user content would be cool, though I think this would also incur translate API usage (at the same rate as actual translation).